### PR TITLE
log entries that are hidden during file listing

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

File: apps/files_external/lib/Lib/Storage/SMB.php [hunk 1]
```diff
@@ -174,15 +183,18 @@ protected function getFolderContents($path) {
 			}
 			return array_filter($files, function (IFileInfo $file) {
 				try {
+					$this->logger->debug('hiding hidden file ' . $file->getName());
```
Comment (blizzz) [inline apps/files_external/lib/Lib/Storage/SMB.php]: doesn't the state depend on the return value of the following method?

**Review by rullzer — APPROVED** [review]
Fine by me :elephant:

**Review by MorrisJobke — APPROVED** [review]

**Comment by skjnldsv** [comment]
@icewind1991 any reason we decide not to show hidden files?

**Comment by icewind1991** [comment]
because showing them would cause inconsistent views between NC and mounting the share in windows and we don't have proper hidden file handling

**Comment by skjnldsv** [comment]
> we don't have proper hidden file handling

What do you mean by that?

We have the show/hide hidden files option in nextcloud, so it's also inconsistent between any smb mount and what is displayed on nextcloud. :thinking:

**Comment by icewind1991** [comment]
that only deals with files starting with `.`, there is no tracking of hiden state in the backend

**Comment by skjnldsv** [comment]
Exactly, and dot files are set as hidden in samba.
So they do not appear.

**Comment by skjnldsv** [comment]
```
smb: \> ls
  .                                   D        0  Mon May 20 23:22:09 2019
  ..                                  D        0  Mon May 20 23:21:40 2019
  .test                              DH        0  Mon May 20 23:22:09 2019

```

**Comment by icewind1991** [comment]
that's samba, with windows smb servers it's different

**Comment by skjnldsv** [comment]
@icewind1991 okay, though this is an issue.
How do we handle this. Do you want me to create an issue?

**Comment by icewind1991** [comment]
I can add an option to show hidden items

**Comment by skjnldsv** [comment]
Awesome @icewind1991 :)
see https://github.com/nextcloud/server/issues/15644

I owe you a beer!

**Pull request merged**